// rebass/dist/util.js
'use strict';

Object.defineProperty(exports, "__esModule", {
  value: true
});
// replace with get?
var idx = exports.idx = function idx(props, obj) {
  var keys = typeof props === 'string' ? props.split('.') : props;
  return keys.reduce(function (a, b) {
    return a && a[b] ? a[b] : null;
  }, obj);
};

var px = exports.px = function px(n) {
  return typeof n === 'number' ? n + 'px' : n;
};

var color = exports.color = function color(props) {
  return function () {
    var n = arguments.length > 0 && arguments[0] !== undefined ? arguments[0] : 'blue';
    var i = arguments.length > 1 && arguments[1] !== undefined ? arguments[1] : '';
    return idx(['colors', baseKey(n) + i], props.theme) || n;
  };
};

var darken = exports.darken = function darken(n) {
  return 'rgba(0, 0, 0, ' + n + ')';
};

var baseKey = exports.baseKey = function baseKey(n) {
  return n.replace(/\d/g, '');
};

var caps = exports.caps = function caps(props) {
  return props.caps ? {
    textTransform: 'uppercase',
    letterSpacing: '.2em'
  } : {};
};

var align = exports.align = function align(props) {
  if (props.left) return 'left';
  if (props.center) return 'center';
  if (props.right) return 'right';
  if (props.justify) return 'justify';
  return null;
};

exports.default = {
  idx: idx,
  px: px,
  color: color,
  darken: darken,
  baseKey: baseKey,
  caps: caps,
  align: align
};
